packages/block-editor/src/components/inner-blocks/index.js

```diff
@@ -200,11 +200,15 @@ export function useInnerBlocksProps( props = {}, options = {} ) {
 		innerBlocksProps.value && innerBlocksProps.onChange
 			? ControlledInnerBlocks
 			: UncontrolledInnerBlocks;
+	//Dedupe layout classes
+	const allTheClassNames = `${ props.className } ${ layoutClassNames }`;
+	const classNameSet = new Set( allTheClassNames.split( ' ' ) );
+	const dedupedClassNames = Array.from( classNameSet ).join( ' ' );
```

Comment:
Rather than de-duping after the fact, is there a way to ensure that layout classnames are only ever added once? I'm not quite sure how we'd do this, but ideally we'd only be outputting `layoutClassNames` in one place, with preference for the inner blocks wrapper where present. Since `useInnerBlocksProps` occurs after `useBlockProps`, I can't think of a clever way to do that, though 🤔

Reply:
This is related to the comment above: the duplication here happens because we're adding layout classnames to BlockListBlock. If we don't do that, this won't be a problem.

>  I can't think of a clever way to do that, though

Can you think of any non-clever ways? 😅

Reply:
The only non-clever thing I could come up with was to avoid `useBlockProps` and `useInnerBlocksProps` altogether, and create a separate hook that grabs `layoutClassNames` from the block edit context (touched on [in this comment](https://github.com/WordPress/gutenberg/pull/44600#issuecomment-1276860156)).

It would mean saying, something along the lines of, "hey, the other block supports happen automatically, but if you want to use layout, you've got to use the layout hook to grab the stuff you need, and then output it where it should go". If it worked, I'm wondering if something like the following pseudo-code might get us in somewhere in that direction:

```js
// layout classnames hook
function useLayoutClassNames( props ) {
  const { layoutClassNames = '' } = useBlockEditContext();
  let mergedClassnames = '';

  // insert code here to merge props.className with layoutClassNames

  return mergedClassnames;
}

// a block's edit component
function MyBlockEdit( props ) {
  const blockProps = useBlockProps();
  const innerBlocksProps = useInnerBlocksProps( blockProps );  

  // The following gets the layout class names and merges them with the classnames from the provided props
  const layoutClassNames = useLayoutClassNames( innerBlocksProps );

  return <div { ...blockProps }>
    <div { ...innerBlocksProps } className={ layoutClassNames } />
  </div>
```

Not sure if something like that's at all viable, but it means that we'd have a hook for getting / merging layout classnames, and the block's Edit component would then be responsible for where to put them. It would mean that becomes part of the API of the layout support for blocks that want to use it, so not sure how viable it would be, because it wouldn't "just work" in the way that we typically expect block supports to work.

But that's all I could think of for now!